ui/components/app/custom-spending-cap/custom-spending-cap.js

```diff
@@ -190,6 +176,25 @@ export default function CustomSpendingCap({
               titleDetailWrapperProps={{ marginBottom: 2, marginRight: 0 }}
               allowDecimals
             />
+            <Box
+              width={BLOCK_SIZES.MAX}
+              marginLeft="auto"
+              paddingRight={2}
+              paddingBottom={2}
+              textAlign={TEXT_ALIGN.END}
+              className="custom-spending-cap__max"
+            >
+              <button
```

Comment:
@georgewrmarshall Do we prefer to use our custom `Button` component in all cases?

Reply:
Yes! Looks like the intention was to use it because it has a `type="link"` prop

Reply:
Hey @darkwing and @georgewrmarshall, I understand what you are saying, but I used `<button />` because when I used `Button` component, `type="link"` prop has his own css styles that I don't need here for `Max` button. 

You can check in `buttons.scss` file what styling is used for `.btn-link`. There are `hover`, `font-size (@include H4)` and `color` different from this that I am using here.

If I used `Button` component, then I will need to add `!important` in `.custom-spending-cap__max--button` class, because it will used styles from `.btn-link` class. I think that `!important` is not good to write in css files.

What is you opinion on this @georgewrmarshall and @darkwing?

Reply:
I think in this case the design is breaking some design system conventions and that the button should be the primary blue color rather than the grey color? @Akatori-Design do you have thoughts on this?

Reply:
@georgewrmarshall I followed design from Figma. [Here](https://www.figma.com/file/n0xMnZqQz4w9Sfa3ZdhWUi/Consentful-experience-for-token-allowances?node-id=2007%3A15726&t=KfwUH6o61gipRzey-0) is that design for `CustomSpendingCap` component and `Max` button in the input field. If @Akatori-Design thinks that the `Max` button should be the primary blue, I will make changes.

Reply:
@VSaric yup your correct you're code follows the design very well! Nice work on that but I think in this particular instance the design is breaking some design system conventions. We should have caught this at the design stage so apologies that I am only catching it now. We don't have any grey colored buttons or links that include text. I actually mocked up this spending cap component to test some of our new design system components. In this story I used the `ButtonLink` component or `<Button type="link">` which I think follows our design system better https://metamask.github.io/metamask-storybook/?path=/story/ui-components-component-library-form-text-field-form-text-field-stories-js--custom-label-or-help-text